playground/frontend/playground_components/lib/src/services/analytics/google_analytics4_service/google_analytics4_service_web.dart

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:html'; // ignore: avoid_web_libraries_in_flutter
+import 'dart:js' as js; // ignore: avoid_web_libraries_in_flutter
+
+import 'package:meta/meta.dart';
+
+import '../events/abstract.dart';
+import 'google_analytics4_service.dart';
+
+final _urlTemplate = Uri.parse('https://www.googletagmanager.com/gtag/js');
+const _eventNameParam = 'eventName';
+
+/// The global JS function to submit analytics data.
+const _function = 'gtag';
+
+@internal
+GoogleAnalytics4Service createGoogleAnalytics4Service({
+  required String propertyId,
+}) =>
+    GoogleAnalytics4ServiceWeb(propertyId: propertyId);
+
+/// Submits data to a Google Analytics 4 property using JavaScript.
+class GoogleAnalytics4ServiceWeb extends GoogleAnalytics4Service {
```

Comment:
Can we see tests for this class?  I see on the GetIt README, the author writes, "I missed the ability to easily switch the implementation for a mocked version without changing the UI".  It seems to suggest that using GetIt seems to provide the ability, as the author writes on the README, "make tests easier".